akka-bench-jmh/src/main/scala/akka/actor/AffinityPoolComparativeBenchmark.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2014-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.BenchmarkActors._
+import akka.actor.ForkJoinActorBenchmark.cores
+import com.typesafe.config.ConfigFactory
+import org.openjdk.jmh.annotations._
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.Throughput))
+@Fork(1)
+@Threads(1)
+@Warmup(iterations = 10, time = 5, timeUnit = TimeUnit.SECONDS, batchSize = 1)
+@Measurement(iterations = 10, time = 15, timeUnit = TimeUnit.SECONDS, batchSize = 1)
+class AffinityPoolComparativeBenchmark {
```

Comment:
@patriknw 

According to my expectations, affinity would show through when actors maintain a certain amount of state. If we just have Ping Pong of case objects, it is quite hard to notice the benefit of not jumping tasks between threads. At least thats how I see it. Therefore I wrote a little bit of different benchmark in 15d4b57df14fe4e354383ad0edfde280fbdf3682 which mimics having 8 actors that act as services, maintaining different state as a in memory map. We round robin and query these actors and measure what is the throughput. I have used your printProgress routing and here is what the results look like at throughput of 1: 

```
Affinity Dispatcher
[info] Iteration   7:   
[info]   720000 messages by 8 actors took 86 ms, 8.29 M msg/s
[info]   720000 messages by 8 actors took 267 ms, 2.69 M msg/s
[info]   720000 messages by 8 actors took 89 ms, 8.03 M msg/s
[info]   720000 messages by 8 actors took 90 ms, 7.93 M msg/s
[info]   720000 messages by 8 actors took 80 ms, 8.93 M msg/s
[info]   720000 messages by 8 actors took 81 ms, 8.80 M msg/s
[info]   720000 messages by 8 actors took 81 ms, 8.84 M msg/s
[info]   720000 messages by 8 actors took 82 ms, 8.77 M msg/s
[info]   720000 messages by 8 actors took 87 ms, 8.23 M msg/s
[info]   720000 messages by 8 actors took 88 ms, 8.09 M msg/s
[info]   720000 messages by 8 actors took 85 ms, 8.37 M msg/s
[info]   720000 messages by 8 actors took 78 ms, 9.16 M msg/s
[info]   720000 messages by 8 actors took 259 ms, 2.77 M msg/s
[info]   720000 messages by 8 actors took 82 ms, 8.71 M msg/s
[info]   720000 messages by 8 actors took 90 ms, 7.97 M msg/s
[info]   720000 messages by 8 actors took 82 ms, 8.68 M msg/s
[info]   720000 messages by 8 actors took 87 ms, 8.26 M msg/s
[info] 590945.450 ops/s


Fork Join Dispatcher
[info] Iteration   7:   
[info]	 720000 messages by 8 actors took 253 ms, 2.84 M msg/s
[info]   720000 messages by 8 actors took 117 ms, 6.14 M msg/s
[info]   720000 messages by 8 actors took 109 ms, 6.55 M msg/s
[info]   720000 messages by 8 actors took 209 ms, 3.43 M msg/s
[info]   720000 messages by 8 actors took 333 ms, 2.16 M msg/s
[info]   720000 messages by 8 actors took 168 ms, 4.28 M msg/s
[info]   720000 messages by 8 actors took 110 ms, 6.52 M msg/s
[info]   720000 messages by 8 actors took 260 ms, 2.77 M msg/s
[info]   720000 messages by 8 actors took 163 ms, 4.41 M msg/s
[info]   720000 messages by 8 actors took 163 ms, 4.41 M msg/s
[info]   720000 messages by 8 actors took 141 ms, 5.09 M msg/s
[info]   720000 messages by 8 actors took 275 ms, 2.61 M msg/s
[info]   720000 messages by 8 actors took 210 ms, 3.42 M msg/s
[info]   720000 messages by 8 actors took 187 ms, 3.84 M msg/s
[info]   720000 messages by 8 actors took 178 ms, 4.03 M msg/s
[info] 521823.138 ops/s
```

I feel however that there is some problem with how JMH is measuring this whole thing as the results I get are quite contradicting. For example, JMH is giving me `521823.138 ops/s `while the printProgress method says that `720000 messages by 8 actors took 178 ms, 4.03 M msg/s` . Maybe I have some problem with configuration of the benchmark.. I m not really sure. What is even more perplexing is that the whole test harness is almost the same as in `ForkJoinActorBenchmark` and `AffinityPoolComparativeBenchmark`. Any ideas are appreciated.